### DISCUSSION TITLE
getting error while adding dependency in android project

### DISCUSSION THREAD

**Discussion by sujatdut** [ngageoint/tiff-java#48] [Q&A]
While trying to integrate this library. 
As soon as I put this in build.gradle 
dependencies {     implementation 'mil.nga:tiff:3.0.0' }
 I am getting this error
Caused by: org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration$ArtifactResolveException: Could not resolve all files for configuration ':annotations:debugCompileClasspath'.

java.lang.RuntimeException: Failed to transform '/Users/xyz/.gradle/caches/modules-2/files-2.1/mil.nga/tiff/3.0.0/8e904ca2930112ec16c8d5df104323ad742d8c8a/tiff-3.0.0.jar' using Jetifier. Reason: null. (Run with --stacktrace for more details.)

In Gradle.properties I already have
android.useAndroidX=true
android.enableJetifier=true
 Still facing this issue.

can someone pls help

**Comment by bosborn** [comment]
Do you have mavenCentral configured as a repository? [Example](https://github.com/ngageoint/geopackage-android/blob/b46c889e0e162b16da584053b0760cad30589748/build.gradle#L5)

or tried using the `android.jetifier.ignorelist` property?

**Reply by sujatdut** [reply]
Yes I have tried both
pasting the full gradle file here. Please let me know if u see any contradiction.

apply plugin: 'com.android.library'

android {
    compileSdkVersion 30
    defaultConfig {
        minSdkVersion 21
        targetSdkVersion 30
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
    aaptOptions {
        cruncherEnabled = false
    }
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
}

buildscript {
    repositories {
        google()
        mavenCentral()
        mavenLocal()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:7.1.3'
        classpath 'com.github.dcendents:android-maven-gradle-plugin:2.1'
    }
}


allprojects {
    repositories {
        google()
        mavenCentral()
        mavenLocal()
//        maven {
//            url  "http://dl.bintray.com/dimitark/maven"
//        }
        jcenter()

    }
}

dependencies {
    implementation("mil.nga:tiff:3.0.0")
    implementation fileTree(include: ['*.jar'], dir: 'libs')
    testImplementation 'junit:junit:4.13-beta-3'
    androidTestImplementation('androidx.test.espresso:espresso-core:3.3.0-alpha02', {
        exclude group: 'com.android.support', module: 'support-annotations'
    })
    implementation 'com.google.code.gson:gson:2.8.5'
    implementation project(':tifflib')
    implementation 'androidx.recyclerview:recyclerview:1.2.1'
    implementation 'androidx.appcompat:appcompat:1.4.0-alpha02'
    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
}

**Reply by sujatdut** [reply]
These are the errors I am getting
java.lang.RuntimeException: Failed to transform '/Users/sujatdut/.gradle/caches/modules-2/files-2.1/mil.nga/tiff/3.0.0/8e904ca2930112ec16c8d5df104323ad742d8c8a/tiff-3.0.0.jar' using Jetifier. Reason: null. (Run with --stacktrace for more details.)
org.gradle.internal.execution.ExecutionException: Execution failed for JetifyTransform: /Users/sujatdut/.gradle/caches/modules-2/files-2.1/mil.nga/tiff/3.0.0/8e904ca2930112ec16c8d5df104323ad742d8c8a/tiff-3.0.0.jar.
org.gradle.api.artifacts.transform.ArtifactTransformException: Failed to transform artifact 'tiff.jar (mil.nga:tiff:3.0.0)' to match attributes {artifactType=android-classes, org.gradle.usage=java-runtime-jars}
